### ISSUE TITLE
restify response takes an additional 100ms from audit logger latency? 

### ISSUE THREAD

**Issue by terencechow** [restify/node-restify#1577]
- [x] Used appropriate template for the issue type
- [x] Searched both open and closed issues for duplicates of this issue
- [x] Title adequately and _concisely_ reflects the feature or the bug

# Bug Report

I am running a kubernetes cluster with 2 restify servers and 1 mysql service. These are run locally with minikube using virtualbox. 1 server is my API and another is my backend which speaks to the local mysql database.

In my restify server I use the audit logger which tells me latency times of a request. Looking at the code of restify, the audit logger tracks the latency of a request from when it is received to the moment it is sent back. Therefore it isn't the moment the Client has received the full response, but the time from when the API has received a request, to when it has started to send a response back.

I'm using Postman as my client. When I use Postman to make a request to my API it says the latency is around 300 ms. However my audit logger from the API says it is only 200ms. 100ms seems like a big difference between my API and Postman, considering this is all within the same computer.

What could cause this difference in latency? Note that the audit logger in my backend shows 100ms, but the audit logger of my API (which does close to 0 processing and is mostly a pass through layer), is showing 200ms, so it seems like there is another 100ms added between servers. Again, 100ms for a request on the same computer seems fairly large no?

In short, I'm seeing an addition 100ms - 200ms for each request on the same computer which seems extremely odd. What could be the cause of this?

## Restify Version
6.3
## Node.js Version
8.9.1
## Expected behaviour
expect audit latency to match response latency
## Actual behaviour
response latency is 100ms+ more than audit latency even on the same computer
## Cause
Not sure

## Are you willing and able to fix this?
I can help debug if possible but not sure I can fix

**Comment by terencechow** [comment]
It appears to occur without kubernetes / minikube. Using process.hrtime I measured 1 route on the backend which just calls mysql. That query takes 60 ms on average from an Apache benchmark of 1000 requests. Literally the next line is `res.send(results)` and yet the audit logger logs something from 200-600 ms when I know my database query isn't the issue. Why might restify be having these problems?

**Comment by DonutEspresso** [comment]
Hi @terencechow, without knowing all the details of your set up it can be hard to say. Payload size can certainly play a part of it - if you are streaming large objects back in the response (say, MBs) that could certainly account for it. 

If possible, I'd recommend setting up a barebones server with nothing but hello world and begin testing from there to rule out any network/environment issues.

**Comment by retrohacker** [comment]
Hey @terencechow,

Closing this as stale for now. If we can get a repro case and some more context, that would help! :heart:
